code/datums/components/ai_leadership.dm

```diff
@@ -0,0 +1,291 @@
+/datum/component/ai_leadership/proc/FollowLeader(mob/living/L)
+	if(ishostile(parent))
+		var/mob/living/simple_animal/hostile/H = parent
+		walk_to(L, parent, 2, H.move_to_delay - 1.5)
+	if(iscarbon(parent))
+		var/mob/living/carbon/H = parent
+		walk_to(L, parent, 2, H.cached_multiplicative_slowdown - 1.5)
+	else if(ishostile(L))
+		var/mob/living/simple_animal/hostile/R = L
+		if(!R.target)
+			walk_to(R, parent, 2, R.move_to_delay)
+	else
+		return FALSE
+	return TRUE
+
+//Orders all troops to follow the leader.
+/datum/component/ai_leadership/proc/Regroup()
+	for(var/i in followers)
+		FollowLeader(i)
+
+	/*-----------------
+	|~ System Checks ~|
+	-----------------*/
+
+/*Determines if the creature is recruitable
+	based on their current status and faction.*/
+/datum/component/ai_leadership/proc/Recruitable(mob/living/L)
+	//Are we at maximum followers
+	if(followers.len >= unit_amount)
+		return FALSE
+	//Are they dead?
+	if(L.stat == DEAD)
+		return FALSE
+	//Player controlled, they do not recognize our authroity.
+	if(L.client)
+		return FALSE
+	//If we are a hostile do these checks
+	if(ishostile(parent))
+		var/mob/living/simple_animal/hostile/H = parent
+		//Are they the same faction as us?
+		if(!H.faction_check_mob(L))
+			return FALSE
+		//If they currently have a target let them finish up.
+		if(H.target)
+			return FALSE
+	var/recruit_type = CheckUnitType(possible_followers, L)
+	//Are they a type in our possible follower types?
+	if(!recruit_type)
+		return FALSE
+	//If we only have the command capacity to command more than one of them.
+	if(unqiue_team)
+		if(!TeamCheck(L))
+			return FALSE
+	//Are they already a follower of another leader?
+	if(FollowerLedger(L))
+		return FALSE
+	return TRUE
+
+//Register recruit with the system
+/datum/component/ai_leadership/proc/Recruit(mob/living/L)
+	followers += L
+	follower_tracker.RecruitFollower(L)
+	FollowLeader(L)
+
+//Releases the recruit from our command
+/datum/component/ai_leadership/proc/Disband(mob/living/L)
+	followers -= L
+	follower_tracker.DismissFollower(L)
+
+//Is the recruit already recruited in the system?
+/datum/component/ai_leadership/proc/FollowerLedger(mob/living/recruit)
+	if(follower_tracker.FindRecruit(recruit))
+		return TRUE
+	return FALSE
+
+//Check the type system
+/datum/component/ai_leadership/proc/CheckUnitType(list/type_list, mob/living/unit_type)
+	for(var/i in type_list)
+		if(unit_type.type == i)
+			return i
+	return null
+
+//Checks if theres any room for this unit in the team.
+/datum/component/ai_leadership/proc/TeamCheck(mob/living/M)
+	var/many_we_have = CountByTroopType(M)
+	var/our_capacity = 0
+	for(var/i in possible_followers)
+		if(M.type == i)
+			our_capacity = possible_followers[i]
+			break
+	if(our_capacity <= many_we_have)
+		return FALSE
+	return TRUE
+
+/*Used to be count by type until i realized count by
+	type includes subtypes when i need a strict type.*/
+/datum/component/ai_leadership/proc/CountByTroopType(mob/living/M)
```

Comment:
This would count by a mob's type, as opposed to a simple typepath, meaning it'd need a reference mob to compare from.

Reply:
What?

Reply:
You need an actual instance of a mob as opposed to just the path in this proc.

Reply:
Okay so i did this weird. The count by troop type apparently is counting how many of the unit being scanned are in the team and if we should just ignore that unit.

Reply:
I need to fix this.